lib/Parse/ParseGeneric.cpp

```diff
@@ -401,6 +405,8 @@ ParserStatus Parser::parseGenericWhereClause(
       break;
     }
     HasNextReq = consumeIf(tok::comma);
+    IsEndOfList = (Context.LangOpts.hasFeature(Feature::TrailingComma) &&
+                   Tok.is(tok::l_brace));
```

Comment:
Same comment that I have on the swift-syntax PR here: Can we allow trailing comma in inheritance clauses because `associatedtype` declarations don’t have a  clear terminator?